frontend/lib/identification/card_detail_view/verification_code_view.dart

```diff
@@ -62,7 +61,15 @@ class _VerificationQrCodeViewState extends State<VerificationQrCodeView> {
                     Padding(
                       padding: EdgeInsets.all(padding),
                       child: QrImage(
-                        data: encodeVerificationCardDetails(VerificationCardDetails(widget.cardDetails, otpCode.code)),
+                        data: const Base64Encoder().convert(
+                          QrCode(
+                            dynamicVerifyCode: DynamicVerifyCode(
+                              info: activationCode.info,
+                              pepper: activationCode.pepper,
+                              otp: otpCode.code,
+                            ),
+                          ).writeToBuffer(),
```

Comment:
I think we should move this convert out of the widget, as it is more application/persistance logicy.